zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -939,6 +949,16 @@ impl Chain {
             .collect()
     }
 
+    /// Returns the Orchard note commitment subtree of the tip of this [`Chain`],
+    pub fn orchard_subtree_for_tip(&self) -> Option<NoteCommitmentSubtree<orchard::tree::Node>> {
+        if !self.is_empty() {
+            let tip = self.non_finalized_tip_height();
+            Some(self.orchard_subtree(tip.into())?)
+        } else {
+            None
+        }
```

Comment:
```suggestion
        let (&index, data) = self.orchard_subtrees.last_key_value()?;
        Some(data.with_index(index))
```

Reply:
This is a behaviour-changing suggestion:
https://github.com/ZcashFoundation/zebra/pull/7636/files#r1340711814